internal/dataplane/configfetcher/config_fetcher.go

```diff
@@ -83,10 +84,18 @@ func (cf *DefaultKongLastGoodConfigFetcher) TryFetchingValidConfigFromGateways(
 			continue
 		}
 
-		if status.ConfigurationHash != sendconfig.WellKnownInitialHash {
+		var kongVersion semver.Version
+		kongVersionStr, getVersionErr := client.GetKongVersion(ctx)
+		if getVersionErr != nil {
+			errs = errors.Join(errs, fmt.Errorf("failed to fetch configuration from %q, cannot get Kong version: %w", client.BaseRootURL(), getVersionErr))
+			continue
+		}
+		kongVersion, getVersionErr = semver.Parse(kongVersionStr)
+
+		if status.ConfigurationHash != sendconfig.WellKnownInitialHash && getVersionErr == nil {
```

Comment:
I know it is very unlikely to happen but to make it bulletproof I think we should check the error first and append it to errs if `!= nil`.

Reply:
removed this part because we do not need `KongVersion` in struct of `KongState`.